atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -71,9 +74,32 @@ def run(self):
 
         self.workflow.pulled_base_images.add(pulled_base)
 
-        if not base_image.registry:
-            response = self.tasker.tag_image(base_image_with_registry, base_image, force=True)
-            self.workflow.pulled_base_images.add(response)
-            pulled_base = response
+        # Attempt to tag it using a unique ID. We might have to retry
+        # if another build with the same parent image is finishing up
+        # and removing images it pulled.
+
+        # Use the OpenShift build name as the unique ID
+        unique_id = get_build_json()['metadata']['name']
+        base_image = ImageName(repo=unique_id)
+        self.workflow.builder.set_base_image(base_image)
```

Comment:
Unsuccessfull tagging would cause exception and fail the build, but it does make sense to move it below the `for` loop to separate it from actual tagging part of the code